base/TorqueStatorSubsystem.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2011-2023 Texas Torque.
+ *
+ * This file is part of TorqueLib, which is licensed under the MIT license.
+ * For more details, see ./license.txt or write <jus@justusl.com>.
+ */
+package org.texastorque.torquelib.base;
+
+import org.texastorque.torquelib.auto.commands.TorqueExecute;
+
+/**
+ * State based subsystem template.
+ *
+ * @author Justus Languell
+ */
+public abstract class TorqueStatorSubsystem<T extends TorqueState> implements TorqueSubsystem {
+    protected T desiredState;
+    protected T lastState;
+
+    protected TorqueStatorSubsystem(final T desiredState) {
+        this(desiredState, desiredState);
+    }
+
+    protected TorqueStatorSubsystem(final T desiredState, final T lastState) {
+        this.desiredState = desiredState;
+        this.lastState = lastState;
+    }
+
+    public final void setState(final T state) {
+        this.desiredState = state;
+        onStateChange();
```

Comment:
It should check if lastState isn't equal to the new State before calling onStateChange.

Reply:
👍